bibtasklets/bst_orcidsync.py

```diff
@@ -49,6 +64,25 @@ def orcid_data_to_sync():
             yield orcid.decode('utf8'), token.decode('utf8'), user_info['email'].decode('utf8'), name
 
 
+def count_orcid_data_to_sync():
+    """
+    Count the number of ORCID tokens (and relative data) to be synch'ed (and
+    thus sent to Redis). This is used at Lab's side to check weather the right
+    number of token has been processed.
+
+    Returns (int): a number.
+    """
+    query = """
+        SELECT count(*) FROM (
+            SELECT personid, count(*) AS cnt FROM inspirehep.aidPERSONIDDATA WHERE tag='uid' AND personid IN (
+                SELECT personid FROM inspirehep.aidPERSONIDDATA NATURAL JOIN inspirehep.aidTOKEN WHERE tag='extid:ORCID'
+            ) GROUP BY personid
+        ) AS Q1;    
+    """
+    count = run_sql(query)
+    return int(count[0][0])
```

Comment:
that's somewhat fragile if the db query failed (intermittent outage, etc. yesterday I got an *out of memory* error on a query against `aidPERSONIDPAPERS`), but then it's probably wise the abort with exception anyhow. so I won't insist on try/except around it.

Reply:
here's a question: What ORCID profile is synced to when there are multiple ORCIDs for the same personid? This is possible in ORCID. currently there is only one such personid in Inspire:
```
|   419277 | extid:ORCID     | 0000-0001-8653-3761 | NULL | NULL | NULL | 2016-08-24 15:41:01 | NULL                                                                                                                 |
|   419277 | extid:ORCID     | 0000-0001-9267-1693 | NULL | NULL | NULL | 2017-10-04 02:10:21 | NULL                                                                                                                 |
|   419277 | uid             | 3877                |    0 |    0 |      | 0000-00-00 00:00:00 | NULL                                                                                                                 |```

Reply:
meanwhile Heath cleaned up that profile, so there's only one orcid left. they appeared to belong to completely different names anyhow.  so the question is moot.

Reply:
Legacy should not allow multiple ORCID by design. So the source of the problem is there.
Anyhow: legacy would push both ORCID profiles to Redis. Labs would read them, but store only the first ORCID per email address and then push with that ORCID.